### PR TITLE
chore: update atlas-sdk to v20231001001

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Update to latest atlas-sdk version https://github.com/mongodb/atlas-sdk-go/releases/tag/v20231001001.0.0

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [ ] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: mongodbatlas/resource_mongodbatlas_project_api_key.go [hunk 1]
```diff
@@ -244,7 +244,7 @@ func resourceMongoDBAtlasProjectAPIKeyUpdate(ctx context.Context, d *schema.Reso
 
 	if d.HasChange("description") {
 		newDescription := d.Get("description").(string)
-		if _, _, err := connV2.ProgrammaticAPIKeysApi.UpdateApiKeyRoles(ctx, projectID, apiKeyID, &admin.CreateAtlasProjectApiKey{
+		if _, _, err := connV2.ProgrammaticAPIKeysApi.UpdateApiKeyRoles(ctx, projectID, apiKeyID, &admin.UpdateAtlasProjectApiKey{
```
Comment (andreaangiolillo) [inline mongodbatlas/resource_mongodbatlas_project_api_key.go:247]: is this change wanted? we don't have the create `CreateAtlasProjectApiKey` function anymore?

Comment (AgustinBettati) [reply mongodbatlas/resource_mongodbatlas_project_api_key.go:247]: Just a naming change that is aligned with the operation. The structure of `admin.CreateAtlasProjectApiKey` and `admin.UpdateAtlasProjectApiKey` are identical.

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Comment by andreaangiolillo** [comment]
@AgustinBettati Could you check that errors are caused by cloud-dev?

**Comment by AgustinBettati** [comment]
> @AgustinBettati Could you check that errors are caused by cloud-dev?

Confirmed it is cloud-dev related, have reach out to the team.

**Review by lantoli — APPROVED** [review]

**Comment by AgustinBettati** [comment]
cloud-dev stable again, all 🟢

**Pull request merged**